packages/core-typings/src/IRoom.ts

```diff
@@ -189,6 +189,9 @@ export interface IOmnichannelGenericRoom extends Omit<IRoom, 'default' | 'featur
 
 export interface IOmnichannelRoom extends IOmnichannelGenericRoom {
 	t: 'l';
+	omnichannel?: {
```

Comment:
@sampaiodiego

Reply:
I got lost here, are we adding an optional property `omnichannel` to an `omnichannel` room? what is the advantage of it? force everyone to do "Optional chaining" everywhere?
like `room.omnichannel?.predictedVisitorAbandonmentAt` ?


right now that we've managed to type pretty much everywhere, and we know we're dealing with omnichannel rooms in the code, are we going to add optional structures everywhere?

Reply:
I would ask to stay "shallow"  as the other 100 props we have for omnichannel and other room typews

Reply:
Yeah, I agree, the nested chaining is obviously not required here, especially since this "omnichannel" prop only has a single field "predictedVisitorAbandonmentAt". But since I wasn't using this prop anywhere in this fix, I didn't modify the logic there, rather just updated our types. However, if you guys are okay with including those migrations here do let me know and I can update this PR accordingly - or if you want me to create a separate PR for it, that works too :smiley: 

Wrt the type is optional, that is actually intentional as this prop will only be set under specific cases even in omnichannel rooms so it's not a generic field which all omnichannel rooms would have, hence it's marked optional.

Reply:
I'd prefer doing that migration on other PR.

As a side note: `omnichannel.xxx` is a really old prop, not being added by the PR. Someone decided to put an omnichannel key with keys inside. All other omni props are top level. This is a design mistake and it indeed looks ugly, but migrating things should be OOS for this PR.